### ISSUE TITLE
Nested Array, get value from Parent array index

### ISSUE THREAD

**Issue by sriks316** [json-editor/json-editor#271]
Hello,

This is my example JSON schema and input:
https://rawgit.com/json-editor/json-editor/master/docs/demo.html?schema=N4IgLglmA2CmIC4QAVYCcDOB7AdiANOAJ4AO8SWARgFawDGYBIJaWZaksGioZY3CUGFLkQAQzRoxRJgDMsaALZjGSMGMpwmkGKNT8mUWIoFCRiEFVoNtULUn1MWbdJ1PEyPD6Ixg0EHABzJlgcAFdFRABtEDoVJgATLGDCSgg0BKY0WBJILUIsMAALdBAAXQBfQhwxRXIzTyRffyCQKpAcXABaPndhRvFJaUMwYz7zChp6RkIdexAAOW79bkJndjcvGrqvfp8/AOCK45PCBNhZMTDoVSiG0SSUjtrRAHUxG9KKyuOgA=&value=N4IgDgpgLgziBcBtUUCekEgCYHsDmIANCAHYCGAthJgOpkA2UEATkaTiQLSSwLKmVq8ECAC+hUOSqYxAXXEp0QkAGMyUNlOUAzCPQCWAD00du0OEkmCZCgdOFzR8oA==&lib_switcher=&prompt_before_delete&theme=bootstrap3&iconlib=fontawesome4&object_layout=grid&show_errors=interaction&disable_edit_json&disable_collapse&disable_properties&disable_array_reorder&disable_array_delete_all_rows&disable_array_delete_last_row

I have an array which has another array.
So the Pets array has a name, non-pets array has a name attribute as well.
Basically, I need to inherit the name from the parent array corresponding to it's index.
So from the example below:
All "names" inside non-pets for dogs should be "Walter"
All "names" inside non-pets for cats should be "should be felix"
Which means, the value needs to be inherited from corresponding parent index value.
Is that doable?
Eg: 
```
{
  "pets": [
    {
      "type": "dog",
      "name": "Walter",
      "non-pets": [
        {
          "name": ""
        },
        {
          "name": ""
        }
      ]
    },
    {
      "type": "cat",
      "name": "felix",
      "non-pets": [
        {
          "name": ""
        },
        {
          "name": ""
        }
      ]
    }
  ]
}
```

**Comment by sriks316** [comment]
Should I add more examples to explain what I am trying to achieve?

**Comment by schmunk42** [comment]
Sorry, I don't have an immediate solution. Maybe a [watcher](https://github.com/json-editor/json-editor#listen-for-changes) or [dependencies](https://github.com/json-editor/json-editor#dependencies) could help...
